Source/WebCore/platform/graphics/texmap/BitmapTexture.cpp

```diff
@@ -106,14 +106,26 @@ void BitmapTexture::reset(const IntSize& size, OptionSet<Flags> flags)
 
 void BitmapTexture::updateContents(const void* srcData, const IntRect& targetRect, const IntPoint& sourceOffset, int bytesPerLine)
 {
-    // We are updating a texture with format RGBA with content from a buffer that has BGRA format. Instead of turning BGRA
-    // into RGBA and then uploading it, we upload it as is. This causes the texture format to be RGBA but the content to be BGRA,
-    // so we mark the texture to convert the colors when painting the texture.
 #if CPU(LITTLE_ENDIAN)
-    m_colorConvertFlags = TextureMapperFlags::ShouldConvertTextureBGRAToRGBA;
+    updateContents(GraphicsContextGL::DataFormat::BGRA8, srcData, targetRect, sourceOffset, bytesPerLine);
 #else
-    m_colorConvertFlags = TextureMapperFlags::ShouldConvertTextureARGBToRGBA;
+    updateContents(GraphicsContextGL::DataFormat::ARGB8, srcData, targetRect, sourceOffset, bytesPerLine);
 #endif
+}
+
+void BitmapTexture::updateContents(GraphicsContextGL::DataFormat pixelFormat, const void* srcData, const IntRect& targetRect, const IntPoint& sourceOffset, int bytesPerLine)
```

Comment:
I would put the pixelFormat the last parameters, or at least after the data. Maybe it would be better to force callers to always pass the format and keep a single function.

Reply:
Agree. If you don't want to touch all the places where updateContents is used, you can use parameter with the default value BGRA, and only the nicosia buffer will pass RGBA.

Reply:
But then we have to change the default value based on, if we're on big/little endian, no?
Because as I understood it's BGRA on little-endian, ARGB on big-endian (for Cairo buffers).

Reply:
> But then we have to change the default value based on, if we're on big/little endian, no? Because as I understood it's BGRA on little-endian, ARGB on big-endian (for Cairo buffers).

Or we can separate the format from whether we're big or little endian. Use BGRA or RGBA for the format. What needs to take the endianess into account is the color conversion flag. That's were you need to set BGRAToRGBA on big endian or ARGBToRGBA on little endian if the format is BGRA.